### ISSUE TITLE
Support maximum community size

### ISSUE THREAD

**Issue by TomKellyGenetics** [TomKellyGenetics/leiden#8]
A "max_comm_size" (default of 0, i.e, any) argument was recently added to leidenalg 0.8.2.

```
  max_comm_size : non-negative int
    Maximal total size of nodes in a community. If zero (the default), then
    communities can be of any size.
```

It would be possible to pass this argument from R if inputs are added. This would require a (recent) version of leidenalg to be installed. 

It may be possible to check whether it is supported in python to retain backwards compatibility but this would complicate things.

```
leidenalg <- import("leidenalg", delay_load = TRUE)
self.optimiser = leidenalg$Optimiser()
missing(self.optimiser$max_comm_size)
```

**Comment by TomKellyGenetics** [comment]
On further consideration, it should be possible to add the argument and pass it with the default of zero for backwards compatibility. The issue is that if it is called on python leidenalg 0.8.1 or earlier (0.8.2 for multiplex methods), it will throw an error.

**Comment by TomKellyGenetics** [comment]
This is now implemented in version 0.3.6 in the "multiplex" branch and version 0.4.0 in the dev branch which supports `igraph::cluster_leiden`. The multiplexed graphs also supported by these upcoming releases. #7 

Passing max_comm_size only occurs if available in Python, maintaining compatibility with previous versions. This does not yet have unit tests or examples in vignettes but is a documented parameter.

A quick check on the examples from the unit tests shows that it works in regular graphs and multiplexed graphs:

```
> table(leiden(snn_graph))

 1  2  3  4  5 
29 21 20 18 12 
> table(leiden(snn_graph, max_comm_size = 0))

 1  2  3  4  5 
29 20 20 19 12 
> table(leiden(snn_graph, max_comm_size = 30))

 1  2  3  4 
28 27 26 19 
> table(leiden(snn_graph, max_comm_size = 5))

 1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 
 5  5  5  5  5  5  5  5  5  5  5  5  5  5  5  5  4  4  4  4  4 
> table(leiden(snn_graph, max_comm_size = 3))

 1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 
 3  3  3  3  3  3  3  3  3  3  3  3  3  3  3  3  3  3  3  3  3  3  3  3  3  3  3  3  3 
30 31 32 33 34 35 
 3  3  2  2  2  1 
```


```
> table(leiden(multiplex_graph,
+                       partition_type = "CPMVertexPartition",
+                       resolution_parameter = 0.1,
+                       seed = 9001))

 1  2  3  4  5  6  7  8  9 10 
14 11 10  8  7  5  3  1  1  1 
> table(leiden(multiplex_graph,
+                       partition_type = "CPMVertexPartition",
+                       resolution_parameter = 0.1,
+                       seed = 9001, max_comm_size = 5))

 1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 
 5  5  5  5  5  4  4  4  4  3  3  3  3  3  3  1  1 
```

The argument is passed to internal functions for bipartite #5 graphs but currently the results do not change. It does not seem to be supported by the [Bipartite](https://leidenalg.readthedocs.io/en/stable/reference.html?highlight=bipartite#leidenalg.CPMVertexPartition.Bipartite) function in Python but instead must be passed to the Optimiser function.

I've verified that it is supported in Python. Here we get different results if we define a non-zero value of `max_comm_size`.

```
>>> 
>>> import leidenalg as la
>>> import igraph as ig
>>> 
>>> def name_object(obj, name):
...   obj.__name__ = name;
...   return obj;
... 
>>> bipartite_graph = name_object(
...     ig.Graph.Bipartite([0, 0, 0, 0, 1, 1, 1, 1],
...                        [[0, 4],
...                         [0, 5],
...                         [0, 6],
...                         [1, 4],
...                         [1, 5],
...                         [2, 6],
...                         [2, 7],
...                         [3, 6],
...                         [3, 7],
...                         [3, 5]]),
...               'bipartite_example')
>>> 
>>> optimiser = la.Optimiser()
>>> p_01, p_0, p_1 = la.CPMVertexPartition.Bipartite(bipartite_graph, resolution_parameter_01=0.1);
>>> diff = optimiser.optimise_partition_multiplex([p_01, p_0, p_1], layer_weights=[1, -1, -1]);
>>> diff
16.8
>>>
>>> optimiser.max_comm_size
0
>>> optimiser.max_comm_size = 5
>>> optimiser.max_comm_size
5
>>> p_01, p_0, p_1 = la.CPMVertexPartition.Bipartite(bipartite_graph, resolution_parameter_01=0.1);
>>> diff = optimiser.optimise_partition_multiplex([p_01, p_0, p_1], layer_weights=[1, -1, -1]);
>>> diff
10.4
```

**Comment by TomKellyGenetics** [comment]
Supported in leiden 0.3.6 with the exception of bipartite graphs (#9).
